tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -0,0 +1,273 @@
+        cls.csm_obj = csm_api_factory("rest")
+        cls.csm_soft_delay = HA_CFG["common_params"]["15sec_delay"]
+        cls.num_users = HA_CFG["s3_operation_data"]["iam_users"] + 50
+
+        for node in range(cls.num_nodes):
+            cls.host = CMN_CFG["nodes"][node]["hostname"]
+            cls.username.append(CMN_CFG["nodes"][node]["username"])
+            cls.password.append(CMN_CFG["nodes"][node]["password"])
+            if CMN_CFG["nodes"][node]["node_type"] == "master":
+                cls.node_master_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+            else:
+                cls.node_worker_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+
+        cls.rest_obj = S3AccountOperations()
+
+    def setup_method(self):
+        """
+        This function will be invoked prior to each test case.
+        """
+        LOGGER.info("STARTED: Setup Operations")
+        self.s3_clean = dict()
+        LOGGER.info("Check the overall status of the cluster.")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        if not resp[0]:
+            resp = self.ha_obj.restart_cluster(self.node_master_list[0])
+            assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Cluster status is online.")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        set_type, set_name = self.node_master_list[0].get_set_type_name(
+            pod_name=pod_list[0])
+        LOGGER.info("Got set_type = %s and set_name = %s.", set_type, set_name)
+        resp = self.node_master_list[0].get_num_replicas(set_type, set_name)
+        assert_utils.assert_true(resp[0], resp)
+        self.def_replica = int(resp[1])
+        self.repl_num = int(len(self.node_worker_list) / 2 + 1)
+        self.deply_name = self.node_master_list[0].get_deployment_name(
+            pod_prefix=const.CONTROL_POD_NAME_PREFIX)[0]
+        LOGGER.info("Default replicas for %s is %s and can be scaled to: %s",
+                    const.CONTROL_POD_NAME_PREFIX, self.def_replica, self.repl_num)
+        if not os.path.exists(self.test_dir_path):
+            resp = system_utils.make_dirs(self.test_dir_path)
+            LOGGER.info("Created path: %s", resp)
+        LOGGER.info("Precondition: Create IAM user before soft-failure.")
+        user = self.mgnt_ops.create_account_users(nusers=1)
+        self.s3_clean.update(user)
+        LOGGER.info("Precondition: Create %s iam users for deletion", self.num_users)
+        self.del_users = self.mgnt_ops.create_account_users(nusers=self.num_users)
```

Comment:
This is not needed in all tests. Please make it test specific, not needed here.

Reply:
> This is not needed in all tests. Please make it test specific, not needed here.

You mentioned to make it default in previous comment. I kept it test specific only. Will revert it in last PR.

Reply:
No, default one user is needed in all tests, i had added comment for that. For multi user we have to keep test specific.